pyeudiw/satosa/backend.py

```diff
@@ -0,0 +1,195 @@
+        it's the authentication response from the underlying OP that is redirected from the OP to
+        the proxy.
+        :rtype: Sequence[(str, Callable[[satosa.context.Context], satosa.response.Response]]
+        :return: A list that can be used to map the request to SATOSA to this endpoint.
+        """
+        url_map = []
+        url_map.append(
+            (f"^{self.entity_configuration_url.lstrip('/')}$", self.entity_configuration))
+        url_map.append(
+            (f"^{self.pre_request_url.lstrip('/')}$", self.pre_request_endpoint))
+        url_map.append(
+            (f"^{self.redirect_url.lstrip('/')}$", self.redirect_endpoint))
+        url_map.append(
+            (f"^{self.request_url.lstrip('/')}$", self.request_endpoint))
+        return url_map
+
+    def entity_configuration(self, context, *args):
+        jwk = JWK()
+
+        data = {
+            "exp": 1649590602,
+            "iat": 1649417862,
+            "iss": "https://rp.example.it",
+            "sub": "https://rp.example.it",
+            "jwks": {
+                "keys": [jwk.export_public()]
+            },
+            "metadata": {
+                "wallet_relying_party": self.config['wallet_relying_party']
+            }
+        }
+
+        jwshelper = JWSHelper(jwk)
+
+        return Response(
+            jwshelper.sign(
+                plain_dict=data,
+                protected={
+                    "alg": "RS256",
+                    "kid": "2HnoFS3YnC9tjiCaivhWLVUJ3AxwGGz_98uRFaqMEEs",
+                    "typ": "entity-statement+jwt"
+                }
+            ),
+            status="200 OK"
+        )
+
+    def pre_request_endpoint(self, context, *args):
+        payload = {'client_id': self.client_id,
+                   'request_uri': self.complete_request_url}
+        query = urlencode(payload, quote_via=quote_plus)
+        response = base64.b64encode(
+            bytes(f'eudiw://authorize?{query}', 'UTF-8'))
+
+        return Response(
+            response,
+            status="200 OK",
+            content="text/json; charset=utf8"
+        )
+
+    def redirect_endpoint(self, context, *args):
+        jwk = JWK()
+
+        helper = JWSHelper(jwk)
+        jwt = helper.sign({
+            "jti": "f47c96a1-f928-4768-aa30-ef32dc78aa69",
+            "htm": "GET",
+            "htu": "https://verifier.example.org/request_uri",
+            "iat": 1562262616,
+            "ath": "fUHyO2r2Z3DZ53EsNrWBb0xWXoaNy59IiKCAqksmQEo"
+        },
+            "RS256",
+        )
+
+        response = {"request": jwt}
+
+        return Response(
+            json.dumps(response),
+            status="200 OK",
+            content="text/json; charset=utf8"
+        )
+
+    def request_endpoint(self, context, *args):
+        jwk = JWK()
+
+        helper = JWSHelper(jwk)
+        jwt = helper.sign({
+            "state": "3be39b69-6ac1-41aa-921b-3e6c07ddcb03",
+            "vp_token": "eyJhbGciOiJFUzI1NiIs...PT0iXX0",
```

Comment:
please use a real presentation token with a digital credential in sd-jwt format here